modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/AddObjectFieldKeyCandidatesMVCResourceCommand.java

```diff
@@ -0,0 +1,134 @@
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryLocalService;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.petra.sql.dsl.Column;
+import com.liferay.petra.sql.dsl.Table;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pedro Leite
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + ObjectPortletKeys.OBJECT_DEFINITIONS,
+		"mvc.command.name=/object_definitions/add_object_field_key_candidates"
+	},
+	service = MVCResourceCommand.class
+)
+public class AddObjectFieldKeyCandidatesMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		long objectDefinitionId = ParamUtil.getLong(
+			resourceRequest, "objectDefinitionId");
+		long[] objectFieldsIds = ParamUtil.getLongValues(
+			resourceRequest, "objectFieldsIds");
+
+		List<String> objectFieldLabels = new ArrayList<>();
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				objectDefinitionId);
+
+		User user = _userLocalService.getUser(objectDefinition.getUserId());
+
+		for (Long objectFieldsId : objectFieldsIds) {
+			ObjectField objectField = _objectFieldLocalService.fetchObjectField(
+				objectFieldsId);
+
+			Table<?> table = null;
+
+			try {
+				table = _objectFieldLocalService.getTable(
+					objectDefinitionId, objectField.getName());
+			}
+			catch (PortalException portalException) {
+				SessionErrors.add(resourceRequest, portalException.getClass());
+			}
+
+			Column<?, ?> column = table.getColumn(
+				objectField.getDBColumnName());
+
+			try {
+				long count = _objectEntryLocalService.getObjectEntriesCount(
+					0, objectDefinition, column.isNotNull());
+
+				if (count == 0) {
+					continue;
+				}
+
+				objectFieldLabels.add(objectField.getLabel(user.getLocale()));
+			}
+			catch (PortalException portalException) {
+				SessionErrors.add(resourceRequest, portalException.getClass());
+			}
+		}
+
+		String errorLabel = "";
```

Comment:
You can inline `errorLabel` and `status` like [here](https://github.com/liferay-objects/liferay-portal/blob/1c5dca33fc65a5f5518c04cafacf7045f4f7d4a0/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java#L189-L221).
You can search for `JSONPortletResponseUtil.writeJSON(` or `HashMapBuilder.<String, Object>put(` to see how you can edit this type of JSON Map.